k8s/upgrades/0.4.0-0.5.0/README.md

```diff
@@ -0,0 +1,173 @@
+percona-2503451898-5k9xw                                         1/1       Running   0          7m
+pvc-8cc9c06c-ea22-11e7-9112-000c298ff5fc-ctrl-3477661062-t0pg9   1/1       Running   0          7m
+pvc-8cc9c06c-ea22-11e7-9112-000c298ff5fc-rep-3163680705-4d7x2    1/1       Running   0          7m
+pvc-8cc9c06c-ea22-11e7-9112-000c298ff5fc-rep-3163680705-lbgpc    1/1       Running   0          7m
+
+test@Master:~/openebs/k8s/demo/percona$ kubectl get svc
+NAME                                                CLUSTER-IP       EXTERNAL-IP   PORT(S)             AGE
+kubernetes                                          10.96.0.1        <none>        443/TCP             24h
+maya-apiserver-service                              10.102.159.226   <none>        5656/TCP            9m
+openebs-grafana                                     10.101.147.181   <nodes>       3000:32515/TCP      45s
+openebs-prometheus-service                          10.106.180.138   <nodes>       80:32514/TCP        45s
+percona-mysql                                       10.100.189.43    <none>        3306/TCP            7m
+pvc-8cc9c06c-ea22-11e7-9112-000c298ff5fc-ctrl-svc   10.111.2.96      <none>        3260/TCP,9501/TCP   7m
+
+**Notes** : This also creates a default prometheus configmap which can be upgraded if needed. The prometheus 
+and grafana services are available on the node ports at ports 32514 & 32515 respectively
+
+```
+### STEP-5: UPDATE OPENEBS VOLUME (CONTROLLER AND REPLICA) DEPLOYMENTS 
+
+Obtain the name of the OpenEBS PersistentVolume (PV) that has to be updated
+
+```
+test@Master:~$ kubectl get pv
+NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS    CLAIM                     STORAGECLASS       REASON    AGE
+pvc-8cc9c06c-ea22-11e7-9112-000c298ff5fc   5G         RWO            Delete           Bound     default/demo-vol1-claim   openebs-basic    
+```
+
+Run the script oebs_update.sh by passing the PV as argument
+
+```
+test@Master:~/0.4-0.5.0$ ./oebs_update pvc-01174ced-0a40-11e8-be1c-000c298ff5fc
+deployment "pvc-8cc9c06c-ea22-11e7-9112-000c298ff5fc-rep" patched
+deployment "pvc-8cc9c06c-ea22-11e7-9112-000c298ff5fc-ctrl" patched
+replicaset "pvc-8cc9c06c-ea22-11e7-9112-000c298ff5fc-ctrl-59df76689f" deleted
+```
+**Notes** : This script replaces the replica and controller patch files with the appropriate container names derived from the 
+PV and patches the volume deployments using the ```kubectl patch deployment``` command. 
+In each case, it verifies whether the new images have been rolled out successfully, using ```kubectl rollout status deployment```
+before proceeding to the next step. Post patching, it also deletes the orphaned replicaset of the controller deployment as a 
+woraround for this issue : https://github.com/openebs/openebs/issues/1201
+
+### STEP-6: VERIFY THAT ALL THE REPLICAS ARE REGISTERED AND ARE IN RW MODE
```

Comment:
Can mayactl be used here instead of curl commands to check the status of the volume? or kubectl command itself?

Reply:
Thanks @kmova , will check on this. One alternative is the maya volume stats, but it currently shows status of a single replica. Will confirm if the "Online" status there reflects RW mode of replica (which it should ideally). Also will insert a step to use kubectl command to show whether the deploys are in running state before entering Step-6 .

Reply:
Have raised issue #1209 requesting maya fix for this functionality. Added a step to perform kubectl get pods to ensure pods are up post upgrade before looking for replica mode.